src/main/java/org/opensearch/knn/index/KNNMethod.java

```diff
@@ -77,6 +78,35 @@ public ValidationException validate(KNNMethodContext knnMethodContext) {
         return validationException;
     }
 
+    public ValidationException validateWithData(KNNMethodContext knnMethodContext, TrainingDataSpec trainingDataSpec) {
+        List<String> errorMessages = new ArrayList<>();
+        if (!containsSpace(knnMethodContext.getSpaceType())) {
+            errorMessages.add(
+                String.format(
+                    "\"%s\" configuration does not support space type: " + "\"%s\".",
+                    this.methodComponent.getName(),
+                    knnMethodContext.getSpaceType().getValue()
+                )
+            );
+        }
+
+        ValidationException methodValidation = methodComponent.validateWithData(
```

Comment:
do we have to validate further even if we don't support space type?

Reply:
Even though validation will ultimately fail if space type is not supported, more error messages can be added to the errors based on any potential problems with the method component context